core/common/privdata/membershipinfo.go

```diff
@@ -7,36 +7,98 @@ SPDX-License-Identifier: Apache-2.0
 package privdata
 
 import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/msp"
 	"github.com/hyperledger/fabric/protos/common"
+	mspp "github.com/hyperledger/fabric/protos/msp"
+	"github.com/pkg/errors"
 )
 
 var logger = flogging.MustGetLogger("common.privdata")
 
 // MembershipProvider can be used to check whether a peer is eligible to a collection or not
 type MembershipProvider struct {
+	mspID                       string
 	selfSignedData              common.SignedData
 	IdentityDeserializerFactory func(chainID string) msp.IdentityDeserializer
 }
 
 // NewMembershipInfoProvider returns MembershipProvider
-func NewMembershipInfoProvider(selfSignedData common.SignedData, identityDeserializerFunc func(chainID string) msp.IdentityDeserializer) *MembershipProvider {
-	return &MembershipProvider{selfSignedData: selfSignedData, IdentityDeserializerFactory: identityDeserializerFunc}
+func NewMembershipInfoProvider(mspID string, selfSignedData common.SignedData,
+	identityDeserializerFunc func(chainID string) msp.IdentityDeserializer) *MembershipProvider {
+	return &MembershipProvider{mspID: mspID, selfSignedData: selfSignedData, IdentityDeserializerFactory: identityDeserializerFunc}
 }
 
 // AmMemberOf checks whether the current peer is a member of the given collection config.
 // If getPolicy returns an error, it will drop the error and return false - same as a RejectAll policy.
-func (m *MembershipProvider) AmMemberOf(channelName string, collectionPolicyConfig *common.CollectionPolicyConfig) (bool, error) {
+func (m *MembershipProvider) AmMemberOf(channelName string,
+	collectionPolicyConfig *common.CollectionPolicyConfig) (bool, error) {
 	deserializer := m.IdentityDeserializerFactory(channelName)
+
+	// Do a simple check to see if the mspid matches any principal identities in the SignaturePolicy
+	if collectionPolicyConfig.GetSignaturePolicy() != nil {
+		memberOrgs, err := getMemberOrgs(collectionPolicyConfig.GetSignaturePolicy().GetIdentities(), deserializer)
+		if err != nil {
+			logger.Errorf("Failed getting member orgs from MSP Principals: %s", err)
+			return false, nil
+		}
+		for _, member := range memberOrgs {
+			if m.mspID == member {
+				return true, nil
+			}
+		}
+		return false, nil
```

Comment:
and here? In these 2 cases I would just let `getPolicy` make the decision for us